validation/js/src/main/scala/hmda/validation/dsl/HmdaDsl.scala

```diff
@@ -0,0 +1,27 @@
+package hmda.validation.dsl
+
+import scala.scalajs.js.Date
+
+trait HmdaDsl {
+  def validTimestampFormat[T]: Predicate[T] = new Predicate[T] {
+    override def validate: (T) => Boolean = _.asInstanceOf[AnyRef] match {
+      case s: String =>
+        checkDateFormat(s)
+      case _ => false
+    }
+    override def failure: String = s"invalid timestamp format"
+  }
+
+  def checkDateFormat[T](s: String): Boolean = {
+    try {
+      val year = s.substring(0, 4)
+      val d = Date.parse(s)
+      d.asInstanceOf[AnyRef] match {
```

Comment:
Is the purpose of this (lines 19-22) just to test whether `Date.parse` was successful? I'm surprised there's not a cleaner way to do that.